### ISSUE TITLE
H + CO2 <=> [O]C=O has two occurances under R_Addition_MultipleBond

### ISSUE THREAD

**Issue by alongd** [ReactionMechanismGenerator/RMG-database#95]
In the Chemkin output file I get the following duplicate reactions for H + CO2 <=> [O]C=O

```
! Reaction index: Chemkin #551; RMG #11336
! PDep reaction: PDepNetwork #1126
! Flux pairs: H(5), CHO2(3053); CO2(30), CHO2(3053); 
H(5)+CO2(30)(+M)=CHO2(3053)(+M)                     1.000e+00 0.000     0.000    
    TCHEB/ 1000.000  1600.000 /
    PCHEB/ 0.493     2.961    /
    CHEB/ 6 4/
    CHEB/ 9.976e+00    3.654e-01    -1.313e-03   -5.167e-05  /
    CHEB/ 2.585e-01    9.412e-03    4.366e-04    1.175e-05   /
    CHEB/ 5.590e-03    -1.645e-04   -2.254e-06   4.709e-07   /
    CHEB/ 7.225e-04    -7.087e-06   1.669e-07    1.970e-08   /
    CHEB/ 2.234e-05    -1.396e-06   1.525e-08    -6.190e-09  /
    CHEB/ -1.768e-06   -2.518e-07   -8.667e-09   -1.136e-10  /
DUPLICATE

! Reaction index: Chemkin #557; RMG #13857
! PDep reaction: PDepNetwork #1639
! Flux pairs: CHO2(3053), H(5); CHO2(3053), CO2(30); 
CHO2(3053)(+M)=H(5)+CO2(30)(+M)                     1.000e+00 0.000     0.000    
    TCHEB/ 1000.000  1600.000 /
    PCHEB/ 0.493     2.961    /
    CHEB/ 6 4/
    CHEB/ 8.885e+00    3.654e-01    -1.313e-03   -5.167e-05  /
    CHEB/ 2.827e-01    9.412e-03    4.366e-04    1.175e-05   /
    CHEB/ 2.605e-03    -1.645e-04   -2.254e-06   4.709e-07   /
    CHEB/ 4.510e-04    -7.087e-06   1.669e-07    1.970e-08   /
    CHEB/ -8.878e-06   -1.396e-06   1.525e-08    -6.190e-09  /
    CHEB/ -5.664e-06   -2.518e-07   -8.667e-09   -1.136e-10  /
DUPLICATE
```

From the PDepNetwork files I get respectively:

```
reaction(
    label = 'reaction1',
    reactants = ['H(5)', 'CO2(30)'],
    products = ['[O]C=O(1169)'],
    transitionState = 'TS1',
    kinetics = Arrhenius(A=(16820,'m^3/(mol*s)'), n=1.23, Ea=(52.7723,'kJ/mol'), T0=(1,'K'), comment="""Estimated using template (Cdd_Od;HJ) for rate rule (CO2;HJ)
Multiplied by reaction path degeneracy 2
Ea raised from 48.7 to 52.8 kJ/mol to match endothermicity of reaction."""),
)
```

```
reaction(
    label = 'reaction1',
    reactants = ['H(5)', 'CO2(30)'],
    products = ['O=[C]O(3053)'],
    transitionState = 'TS1',
    kinetics = Arrhenius(A=(0.7734,'m^3/(mol*s)'), n=2.941, Ea=(35.6477,'kJ/mol'), T0=(1,'K'), comment="""Estimated using template (Od_R;HJ) for rate rule (Od_Cdd-Od;HJ)
Multiplied by reaction path degeneracy 2."""),
)
```

For the first reaction, CO2 is recognized as `Cdd_Od`, and for the second as `Od_Cdd-Od`, both in R_Addition_MultipleBond (unrelated: could be nice to have the family name integrated into the PDepNetwork comments, see https://github.com/ReactionMechanismGenerator/RMG-Py/issues/605)

From my understanding, this approximately doubles the reaction rate as both duplicate reactions are accounted for when generating the flux.

**Comment by rwest** [comment]
The two elementary reactions are different. One adds to the C end of the double bond, the other adds to the O end of the double bond, and they have different products, so should both exist and be estimated independently. So far, I don't see a bug.

How this feeds in to the pressure dependence code I am not an expert, but here are my thoughts: Presumably the two elementary reactions are two different entrance channels onto the same potential energy surface. Once on that surface, both entrances could lead to the same well (CHO2). I think you see duplicates because we solve each entrance as a different PDep network (I think this approach is needed to figure out leak fluxes, or something. @jwallen will know the gory details, as for a while I think his algorithm did merge all the networks and solve them together, before reverting to David Matheu's [XMG-Pdep](https://web.anl.gov/PCS/acsfuel/preprint%20archive/Files/49_1_Anaheim_03-04_0845.pdf) approach. I'm not sure when/how networks get merged). 

So what you see might be it functioning as intended: one Chebyshev represents the H adding to the O then isomerizing to CHO2 before being stabilized, and the other Chebyshev represents H adding to the C and being stabilized directly. How to check that the numbers are right, I'm not sure. Do these networks ever merge?

**Comment by alongd** [comment]
Hi, I see I made a mistake above and didn't copy the right reaction from the PDep network.
In the original message above, both reactions from the Chemkin file involve the ```CHO2(3053)``` species, but I mistakenly copied a reaction from the PDep network that yields a different product, ```[O]C=O(1169)```, as @rwest rightly mentioned. Sorry, totally my bad!!

To correct, here are the two PDepNetwork reactions that correspond to the duplicate reactions from the Chemkin file:

network1126_2.py:
```
transitionState(
    label = 'TS6',
    E0 = (-155.857,'kJ/mol'),
    spinMultiplicity = 1,
    opticalIsomers = 1,
)

reaction(
    label = 'reaction1',
    reactants = ['H(5)', 'CO2(30)'],
    products = ['O=[C]O(3053)'],
    transitionState = 'TS6',
    kinetics = Arrhenius(A=(0.7734,'m^3/(mol*s)'), n=2.941, Ea=(35.6477,'kJ/mol'), T0=(1,'K'), comment="""Estimated using template (Od_R;HJ) for rate rule (Od_Cdd-Od;HJ)
Multiplied by reaction path degeneracy 2"""),
)
```

network1639_1.py:
```
transitionState(
    label = 'TS1',
    E0 = (-155.857,'kJ/mol'),
    spinMultiplicity = 1,
    opticalIsomers = 1,
)

reaction(
    label = 'reaction1',
    reactants = ['H(5)', 'CO2(30)'],
    products = ['O=[C]O(3053)'],
    transitionState = 'TS1',
    kinetics = Arrhenius(A=(0.7734,'m^3/(mol*s)'), n=2.941, Ea=(35.6477,'kJ/mol'), T0=(1,'K'), comment="""Estimated using template (Od_R;HJ) for rate rule (Od_Cdd-Od;HJ)
Multiplied by reaction path degeneracy 2"""),
)
```

I'll also correct that in both reactions CO2 is recognized as ```Od_Cdd-Od```, which is OK.
But still, these are identical reactions with the same kinetic coefficients (strange that identical rates came out differently in the Chebyshev fit, though), and both end up in the final output. 

One side note, don't know if related: all of the PDep network files in this run end with "..._1.py" except for "network1126_2.py", which happens to take part in the duplicity here.

**Comment by alongd** [comment]
I should have originally opened this issue in -Py, not -db.
I recently got a similar behaviour (of two identical PDep reactions being generated from different networks). Tracing this problem, I found that each network correctly contains that same reaction, but with different isomers and different P-Dep well-skipping branching. Although confusing, it looks correct.
I'm closing this issue, if there's more interest in this point we could resume the discussion in the -Py repo.
